compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirNotImplementedOverrideChecker.kt

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.descriptors.Modality
+import org.jetbrains.kotlin.descriptors.Visibilities
+import org.jetbrains.kotlin.fir.*
+import org.jetbrains.kotlin.fir.analysis.checkers.FirDeclarationPresenter
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.checkers.getContainingClass
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.fir.scopes.*
+import org.jetbrains.kotlin.fir.scopes.impl.unwrapDelegateTarget
+import org.jetbrains.kotlin.fir.symbols.CallableId
+import org.jetbrains.kotlin.fir.symbols.impl.*
+import org.jetbrains.kotlin.fir.types.*
+import org.jetbrains.kotlin.name.Name
+
+object FirNotImplementedOverrideChecker : FirClassChecker() {
+    // TODO: differentiate members with different annotations, e.g., @Api...(x) @Api...(y)
+    // TODO: differentiate parameter names? e.g., foo(x : T) v.s. override foo(y : T)
+    private object SignaturePresenter : FirDeclarationPresenter {
+        override fun StringBuilder.appendRepresentation(it: CallableId) {
+            append(it.callableName)
+        }
+    }
+
+    private fun FirClass<*>.unsubstitutedScope(context: CheckerContext) =
+        this.unsubstitutedScope(context.sessionHolder.session, context.sessionHolder.scopeSession, withForcedTypeCalculator = true)
+
+    // TODO: Collection extensions are not properly regarded as an override. Should not have a whitelist like this.
+    private val COLLECTIONS = listOf("java.util", "kotlin.collections")
```

Comment:
What about `kotlin.(random|sequences|text|...)`?

Reply:
I heard @dzharkov is working on fixing the way overrides of Collection extensions are populated in scopes. Until then, this is just a stopgap to avoid raising any false alarms. Due to Java interop and type aliases, collections are somewhat special and non-trivial to find overrides I guess? I'm not seeing any false alarms regarding those packages, but it's maybe the case that no tests are out there that exercise extending those packages (or do they have any overridable members?).

Reply:
I see. Some of them have overridable members. But since this is not the long term solution, I am fine with keeping this as it is.

Reply:
Right. I don't like this kind of whitelist either, and I don't want to introduce a checker with false alarms. This is a sweet spot (i.e., minimal whitelist) at this moment. :p